### PR TITLE
Populate submodules of `torch._C` to `sys.modules` recursively

### PR DESCRIPTION
**Pull request by XuehaiPan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #132216

See comment:

https://github.com/pytorch/pytorch/blob/e9d1c262756d883cc597f2791d482aced4e1cb84/torch/__init__.py#L938-L950

This PR recursively sets the submodules in the C extension to `sys.modules` (e.g., `_C._dynamo.eval_frame`).

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

### CODE REVIEW

File: torch/_dynamo/utils.py [hunk 1]
```diff
@@ -2228,7 +2228,6 @@ def nn_module_get_all_hooks(
     check_backward_hooks=False,
     check_state_dict_hooks=False,
 ):
-    reset_code = torch._C._dynamo.eval_frame.reset_code
```
Comment (XuehaiPan) [inline torch/_dynamo/utils.py:2231]: unused

**Comment by ezyang** [comment]
Besides the CI failures, I'm puzzled why you had to change the use sites too

**Comment by XuehaiPan** [comment]
> Besides the CI failures,

Split into a separate PR #132377

- #132377

> I'm puzzled why you had to change the use sites too.

Import statements have better static linter support than assignments.

**Review by ezyang — APPROVED** [review]

**Comment by PaliC** [comment]
We are reverting this PR as its breaking a bunch of windows tests internally

Examples of the errors include: https://pastebin.com/uxUxV2hm

**Comment by ezyang** [comment]
Please merge https://github.com/pytorch/pytorch/pull/132763 into this PR, along with whatever fix is needed

**Comment by malfet** [comment]
@XuehaiPan , @ezyang What is the urgency/impact of landing this change? It breaks some internal Windows builds, which makes me wonder what it actually fixes

**Comment by XuehaiPan** [comment]
@malfet This is not a bug fix but a enchancement. Feel free to revert it if it is not applicable. The motivation for this change is that from-imports have better static analysis support than assignment.

**Pull request closed without merging**